src/System.Private.CoreLib/src/System/RtType.cs

```diff
@@ -4819,20 +4817,9 @@ private enum DispatchWrapperType : int
 
         private static int GetUtf8StringByteLength(void* pUtf8String)
         {
-            int len = 0;
-
-            unsafe
-            {
-                byte* pItr = (byte*)pUtf8String;
-
-                while (*pItr != 0)
-                {
-                    len++;
-                    pItr++;
-                }
-            }
+            const int MaxStringLength = 1024;
```

Comment:
Previously it was unlimited length; but the `SpanHelper.IndexOf` isn't happy with that. Have chosen 1024; is that too small?

Reply:
The risk is that `IndexOf('\0')` may access memory that it is not supposed to access when it is given arbitrary sized buffer. It should not happen today given what the implementation looks like, but it is not a good idea to start spreading subtle assumptions like these throughout the codebase.

For now, I would just call `StubHelpers.strlen` or some other proper strlen variant here (there are several). Consider refactoring of `wcslen`/`strlen` to use fully managed (vectorized, etc.) implementation in separate change.

Reply:
To keep it isolated to one place and easier to refactor/identify? Makes sense.

The comment in the unmanaged code that I removed for the string compare is:
```cpp
// Important: the string in pSsz isn't null terminated so the length must be used
// when performing operations on the string.
```
So it does feel an uncomfortable coupling; though I suppose that's for the other .ctor of `MdUtf8String` which takes length rather than looking for a null termination?

Reply:
>  Consider refactoring of wcslen/strlen to use fully managed (vectorized, etc.) implementation in separate change.

https://github.com/dotnet/coreclr/pull/21729